src/io/config.cpp

```diff
@@ -42,6 +42,19 @@ std::unordered_map<std::string, std::string> Config::Str2Map(const char* paramet
   for (auto arg : args) {
     KV2Map(&params, Common::Trim(arg).c_str());
```

Comment:
Looks like the following warnings in `KV2Map()` function are still cannot be suppressed by passing `verbosity` param because `// define verbosity and set logging level` happens later in the code
https://github.com/microsoft/LightGBM/blob/1cc9f9dcee15586aefdd8271a1c04ad010d3c53a/src/io/config.cpp#L28-L35

Reply:
These warnings come from duplicated parameters, not aliases. I can try to address them as well in this PR if its in the scope.

Reply:
I believe it's related.

Reply:
I agree with you. I believe each warning is helpful, but unfortunately LightGBM users want to completely silence their program (according to the issues they create in the repo).

Reply:
Addressed by having an unordered_map of vectors where we store all parameters for as many times as they're defined, set the verbosity taking the first values of `verbose` and `verbosity` and then de-duplicating with the `KeepFirstValues` function, where warning level logs are printed but after the verbosity has been set, so these can be disabled as well.